File: backend/src/controllers/quiz.controller.js
const quizService = require("../services/quiz.service");

// Tạo Quiz
exports.createQuiz = async (req, res) => {

    try {

        const quiz = await quizService.createQuiz(
            req.user.id,
            req.body
        );

        res.status(201).json({
            success: true,
            data: quiz
        });

    } catch (error) {

        res.status(403).json({
            success: false,
            message: error.message
        });

    }

};
// Tạo quiz + danh sách câu hỏi
exports.createFullQuiz = async (
    req,
    res
) => {

    try {

        const quiz =
        await quizService.createFullQuiz(

            req.user.id,

            req.body

        );

        res.status(201).json({

            success: true,

            data: quiz

        });

    } catch (error) {

        console.error(error);

        res.status(400).json({

            success: false,

            message: error.message

        });

    }

};
exports.getQuizDetail = async (
    req,
    res
) => {

    try {

        const quiz =
        await quizService.getQuizDetail(

            req.user.id,

            req.params.id

        );

        return res.json({

            success: true,

            data: quiz

        });

    } catch (err) {

        return res.status(403).json({

            success: false,

            message: err.message

        });

    }

};
// Lấy danh sách Quiz
exports.getQuizzes = async (req, res) => {

    try {

        const quizzes =
        await quizService.getQuizzesByCourse(

            req.user.id,

            req.params.courseId

        );

        res.json({

            success: true,

            data: quizzes

        });

    } catch (error) {

        res.status(403).json({

            success: false,

            message: error.message

        });

    }

};
//lấy các quiz từ ccs lớp
exports.getMyQuizzes = async (req, res) => {

    try {

        const data =
        await quizService.getMyQuizzes(
            req.user.id
        );

        return res.status(200).json({

            success: true,

            data

        });

    } catch (err) {

        console.error(err);

        return res.status(500).json({

            success: false,

            message: err.message

        });

    }

};
// Cập nhật Quiz
exports.updateQuiz = async (req, res) => {

    try {

        const quiz =
        await quizService.updateQuiz(

            req.user.id,

            req.params.id,

            req.body

        );

        res.json({

            success: true,

            data: quiz

        });

    } catch (error) {

        res.status(403).json({

            success: false,

            message: error.message

        });

    }

};

// Xóa Quiz
exports.deleteQuiz = async (req, res) => {

    try {

        await quizService.deleteQuiz(

            req.user.id,

            req.params.id

        );

        res.json({

            success: true,

            message: "Deleted successfully"

        });

    } catch (error) {

        res.status(403).json({

            success: false,

            message: error.message

        });

    }

};

File: frontend/lib/controllers/quiz_controller.dart
import '../models/quiz_model.dart';
import '../services/quiz_service.dart';
import '../models/quiz_group_model.dart';
class QuizController {

  final QuizService _service =
  QuizService();

  // Lấy danh sách Quiz
  Future<List<QuizModel>> getQuizzes(
      int courseId) async {

    return await _service.getQuizzes(
      courseId,
    );

  }
  // Lấy tất cả Quiz của các lớp đã tham gia
  Future<List<QuizGroupModel>> getMyQuizzes() async {

    return await _service.getMyQuizzes();

  }
  // Tạo Quiz + Question
  Future<bool> createFullQuiz(
      QuizModel quiz) async {

    return await _service.createFullQuiz(
      quiz,
    );

  }
  Future<QuizModel> getQuizDetail(
      int quizId,
      ) async {

    return await _service.getQuizDetail(
        quizId);

  }

  // Cập nhật Quiz
  Future<bool> updateQuiz(
      QuizModel quiz) async {

    return await _service.updateQuiz(
      quiz,
    );

  }

  // Xóa Quiz
  Future<bool> deleteQuiz(
      int quizId) async {

    return await _service.deleteQuiz(
      quizId,
    );

  }

}